modules/flowable-ui-task/flowable-ui-task-rest/src/main/java/org/flowable/app/rest/runtime/RuntimeDisplayJsonClientResource.java

```diff
@@ -128,7 +128,7 @@ public JsonNode getModelJSON(@PathVariable String processInstanceId) {
 
         ProcessInstance processInstance = runtimeService.createProcessInstanceQuery().processInstanceId(processInstanceId).singleResult();
         if (processInstance == null) {
-            throw new BadRequestException("No process instance found with id " + processInstanceId);
+            return getModelHistoryJSON(processInstanceId);
```

Comment:
Hi Martin, this change modifies the behaviour of the getModelJson method. If you need the history json then the history endpoint should be invoked.

Reply:
Done